### ISSUE TITLE
"Error:/bin/sh: haxelib: command not found" when trying build a project using a HXML file

### ISSUE THREAD

**Issue by b005t3r** [HaxeFoundation/intellij-haxe#549]
Plugin: Version: 0.10.1.1 for IDEA 2016.1 through 2016.3.2
IDEA: 2016.3.4

Also tested with:
Plugin: Version: 0.10 for IDEA 14.0
IDEA: 14.0

This is what it output when I hit build:
```
Information:14/02/17 17:07 - Compilation completed with 2 errors and 0 warnings in 91ms
Information:Executing: /usr/local/lib/haxe/haxe /Users/booster/Documents/haxe/hxsl/samples/0_cube/test.hxml 
Error:Compilation failed
/Applications/IntelliJ IDEA.app/Contents/bin
Error:/bin/sh: haxelib: command not found
```

The project compiles and runs just fine with command line tools. Both haxelib and haxe are accessible from the command line.

Contents of HXML file:
```
-cp ../lib
-swf test.swf
-swf-header 1920:1080:60:FFFFFF
--flash-strict
-swf-version 11
-lib hxsl
-main Test
```

It seem to work with Haxe Compiler option selected.

**Comment by cmandlbaur** [comment]
If you're running on macOS try launching from terminal. I've had issues with macOS not passing env vars properly to apps launched from finder.

**Comment by b005t3r** [comment]
Running idea from terminal?

**Comment by EricBishton** [comment]
@b005t3r, @cmandlbaur  is right.  macOS goes out of its way to isolate environments.  You must set the absolute path(s) to haxelib (neko, toolkit) in your Haxe SDK: File->ProjectSettings...->(Left pane)SDKs->(Center pane){Haxe SDK Name}->(Right pane)Haxelib Executable.

In the same dialog, be sure that your haxelib directory is included on the classpath.

**Comment by EricBishton** [comment]
@b005t3r Running IDEA from the terminal _ought_ to work, if the terminal has all of the environment variables set.

**Comment by b005t3r** [comment]
I'll try the terminal trick, however I never had problems like that with any other application. 

As for the paths to haxe and haxelib these were set.

**Comment by cmandlbaur** [comment]
Does that mean that mine is incorrectly configured?

**Comment by EricBishton** [comment]
> I never had problems like that with any other application

Most applications don't require/use applications installed independently.  There may be something we can do here, though:  We can try to duplicate the environment variables that the Haxe installer creates (or have users set them directly in the SDK dialog), and make sure they are set before we fork any compilation processes.

(The trouble is, I don't have a Mac at my disposal any more...)

**Comment by b005t3r** [comment]
OK, the terminal trick works. Of course there are other issues that I'll create entries for later :)

**Comment by b005t3r** [comment]
Did you maybe do anything that could fix this issue in the 10.1.Test.CompilerCompletion build? Because I just noticed I'm launching IDEA normally, not from the Terminal, and it works :)

I recently updated to macOS Sierra, could it be it?

**Comment by EricBishton** [comment]
Hmm.  Could be Sierra, I guess.  In the completion build, I did change how the process used for completion is started and checked, but I didn't add any code to find the Hate environment variables and add them to the path (though I did look into it).

Could it be that the variables were inserted into the master environment, but required a reboot (which you did with the upgrade) in order to actually pick them up into launcher's environment?
